app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,42 @@
+class PostsController < ApplicationController
+    before_action :sign_in?, only: %i[new create destroy]
+
+    def index
+        @posts = Post.all
+    end
+
+    def new
+        @post = current_user.posts.build
+    end
+
+    def create
+        @post = current_user.posts.build(post_params)
+        if @post.save
+            flash[:success] = "Post successfully created"
+            redirect_to postlist_path
+        else
+            flash.now[:error] = "Unable to create post"
+            redirect_to '/login'
+        end
+    end
+
+    def show
+        @post = Post.find(params[:id])
```

Comment:
`@post = Post.find(params[:id])` you are doing this for show and delete and would do it again if you implemented edit and update. consider a private method and set it with a before_action.   Not a big deal here anyway.